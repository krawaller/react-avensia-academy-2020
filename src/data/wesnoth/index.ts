import _unitData from './units.json'
import _base64 from './base64.json'
import { Unit } from './types'

type UnitId = keyof typeof _unitData
export const unitIds = Object.keys(_unitData).sort() as UnitId[]

export const base64 = _base64 as Record<string, string>
export const units = _unitData as Record<UnitId, Unit>
export * from './types'
