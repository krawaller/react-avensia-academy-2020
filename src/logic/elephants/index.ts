export * from './model'
export * from './actions'
export * from './reducer'
export * from './flipMoodInState'
export * from './addElephantToState'
export * from './makeLyrics'
