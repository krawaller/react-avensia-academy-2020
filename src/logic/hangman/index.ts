export * from './getRevealedLetters'
export * from './getNbrOfErrors'
export * from './getSessionStatus'
export * from './newSessionState'
export * from './addGuessToSession'
export * from './actions'
export * from './reducer'
export * from './model'
