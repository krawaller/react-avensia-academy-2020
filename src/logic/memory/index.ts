export * from './getSessionStatus'
export * from './newSession'
export * from './flipTile'
