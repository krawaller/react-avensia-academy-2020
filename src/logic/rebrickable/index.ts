export * from './getAllThemes'
export * from './getSetsForTheme'
export * from './getThemesByParent'
export * from './types'
