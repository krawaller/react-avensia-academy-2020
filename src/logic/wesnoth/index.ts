import _unitData from './units.json'
import _base64 from './base64.json'
import { Unit, RaceName, UnitLibrary } from './types'

export type UnitId = keyof typeof _unitData
export const unitIds = Object.keys(_unitData).sort() as UnitId[]

export const base64 = _base64 as Record<string, string>
export const units = _unitData as Record<UnitId, Unit>
export * from './types'

export const races = Object.values(units).reduce((memo, unit) => {
  if (!memo[unit.race]) {
    memo[unit.race] = {}
  }
  memo[unit.race][unit.id] = unit
  return memo
}, {} as Record<RaceName, UnitLibrary>)
