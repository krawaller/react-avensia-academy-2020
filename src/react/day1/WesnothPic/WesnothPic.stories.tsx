import React from 'react'
import { storiesOf } from '@storybook/react'
import { select } from '@storybook/addon-knobs'
import { WesnothPic } from './WesnothPic'
import { base64 } from '../../../toys/wesnoth'

const picIds = Object.keys(base64).sort()

storiesOf('Day 1/Composition/WesnothPic', module).add('dropdown knob', () => {
  const picId = select('Picture', picIds, 'attacks/axe-deathblade.png')
  return <WesnothPic id={picId} />
})
