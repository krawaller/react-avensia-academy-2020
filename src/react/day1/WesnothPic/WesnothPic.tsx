import React from 'react'
import { base64 } from '../../../toys/wesnoth'

export type WesnothPicProps = {
  id: string
}

const picTypeSizes = {
  attacks: 60,
  units: 72,
  terrain: 30,
  icons: 16,
}

export const WesnothPic = (props: WesnothPicProps) => {
  const type = props.id.split('/')[0] as keyof typeof picTypeSizes
  const size = picTypeSizes[type]
  return (
    <div
      style={{
        backgroundImage: `url(${base64[props.id]})`,
        height: size,
        width: size,
        backgroundRepeat: 'no-repeat',
        display: 'inline-block',
      }}
    />
  )
}
