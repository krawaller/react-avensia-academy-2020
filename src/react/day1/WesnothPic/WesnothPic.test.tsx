import React from 'react'
import { render } from '@testing-library/react'
import { WesnothPic } from './WesnothPic'
import { base64 } from '../../../logic/wesnoth'

const picIds = Object.keys(base64).sort()

describe('the WesnothPic component', () => {
  it('renders as expected', () => {
    const { container } = render(<WesnothPic id={picIds[0]} />)
    expect(container).toMatchSnapshot()
  })
})
