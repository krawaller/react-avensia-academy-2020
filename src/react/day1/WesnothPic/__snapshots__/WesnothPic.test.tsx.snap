// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the WesnothPic component renders as expected 1`] = `
<div>
  <div
    style="background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADwAAAA8CAYAAAA6/NlyAAAQS0lEQVR42u2bCUyVZ9bHC1RkKesgW0TRoMCwDwIybPf2sg2LULggA1wgCPixqGGpBBGjaI2IGve4EavW2FaMWxutmrZa00VNR9u01TY67Uzar0umX9u0M+2knd93njuZFt/70nYsTsnM3OTkwH2X5/3/z3nOc87znnvPfz//KR8vbzf8A3xQ2vsX7nh4uuLu4YKzywQm+XrhM8kTN3dnfP28CQicZP37Fz4e1vMc7r2H+9yccHF1VOcqbRVPr/u4d8I96hrreU7O9+Lufh/33edi1U5Ojnh7e+Lp6Y6rq7P6zqrd3FwJCPBT31u/8/X1Udeo79V31mt8fLzVd9Zj/v6+/7jXt8eU9vBws14j8LABrB6qvOwBZifHY7w/1SqmzAwyswyUmousOiIylAdKCvlNXhZh4SHW78xlxWRlG63n3G9KV/+r66z/t7Q2Wc9JTZtNds79NLc08tsKM2XmYutYWZkG69+/yc0k05ShrrMeT0tPJjvLSGxcJDHRv8RgTOXXyQmUzy2R82Ss0gfkuhLuv9/A3HIzjY3z5B7ZpGf8GqMhnf9pblD3tt6vuChfGdAWsLKAGlj0v42F1TMkJsQpbQtYQI5kY1xJaUEmb109x/C2AfjmM+Bz/vT+Ta5ducBzz5zGy07/OnFp5TlK2wLWsDGupGdhA6eODLG2s5mRn88+fZ/Xrr5EhIvTqICjo8L1AWvYGFeydaCXTWuW0t9WD19/zJtvXOHDd98C4O1br2Jwnqh7nZoCiUm/UtoWsIaNcSMODvY8d3I/PU3VLG+t5cYzx3n69FGe2L+L448NcfbUEVo8XEYFLMFLH7CGjXEjs2LCOTO8h4d6O1jV0cgrpx9j3/qV3HrzKof2beORvVupc3cZ1aUlUuu7tIaNcSPLu5rZtaGf/r52BpZ3kGdMpbwoi77FzXS2WTBlJBHspB93JFIrTErbAtawMU7c2YGTBzbTLUGrpmIOBYZkCVxt/P61K/CXP/Hmqy+yYU0v60yputfL8qTWZKVtAWvYGBcSHxXGqYNbKC4wkWNMpqemlOENA/DXD7l54xrPnnmCi0/LHE6JH9XCkojoW1jDxriQeXOL2bmujxZLKb2djXQ3zGX3gwuA93np4jNcfemcgD7K9obKUQFL9qUPWMPGuJAda/tYtnghw3sG2bZhGZ3FObx56Sk++uBtzp89xfHDBxk+tJt1ETN1r5fsS6WcStsC1rDxs4u3pwfHD26nvbmGPQJ25eIm2mvL+Pyj61x/7TJPnRhmaMcmLp17jEjHCbr3kJRT5dlK2wLWsHHXxcXRDrvvOV4i6eSebatZWFPCvOpCFpTl8mCtmT/euCDWPcLOzQNsHuzn6rNHiJ5or3sPybNVcaG0LWANG3dVogKdObeqhLacGXi7ONgcnzjRkYd3PISlvIgcWXaaLEV0WUoozk2no6WGgf7FPLJzkP7uhXDhJB76ubQqLlRFpbQtYA0bd0Uc7O3xlSlTmzyJRztNvHqshxcOtRMW6HHbeaaM2ezbvZrqknzqyvJozcmgriSbttw0bly7AH99j2dPHqKjsYKlhRmjjicBWJWR+suSho0xFy9PTwpys4gIm0FpvA/LzOE8u7mWvQvSiQj8bkw7OztW9nXK2lvPgiYLlnwDJYkxNGamMbCyC775gptvvMiXn70tldIJBo2aJUkTpaV21o/SGjbGVBwdHSkpzGOFpIfm4jwyZoWyuiaBp3Y2YAx1x26ES6bOjmfnlrW0t9bQVJSLKSKE+opCFrbU88bVZ4BPuPH677h5/WUJXAfojwkZdVypndWGgdK2gDVsjKkkJ86ipaFCKpzrfPHRW1bLzTHEUpg0HQ835xEu6MOe7RsYXLWYJuNsKhOiqBZXbpBg1d3Zwtd/fhe+eo/LL1xky7qHOHJwK74OdqOOK2my2iXRLx40bIyZKOvFSsZ08ugBXvnd8/D1pzx1/BClYr24iBmkx0eqIGU9d1FrE9s29lGbnU5K8GRqzHnUzc2nt3sB1y6dBj4VsM/x5LFhLj19lPOdjSO8Qz9oydaQftDSsDGGgO1wc3Vl9dIuyZaa2LlpDe3zq2isKRU3tbCgcS5ZhkRioyPFaqtorC2nODWRjOQYSvLSqauawyVZevjyD3z1yducO3WSxw/s5fWXTrHHlKYdT7ssqf0w/WVJw8aYyURJCoJkwKiQqZJA9AnwFuaKm+bFxVBWlEleVAQxAnbr+n7aW2opzctgjkTj0gIDFcVZdC2YxxuXT3HmxEGOSP07JC7fv2QRFaW5OAqZ3ze27IepTUClbQFr2BgziQxypTJlOjMn+xI2NYA5qbHUVRZTmpaEcVYMGWGhEqAaWCplYFNNMfXleVQWZ2Ipy2HJglqO797M4S0PceKRLVSX5bOosQZDcjzTxeU1Y+kGLdn51A9aGjbGRDxdHWlK96d3TgjnHirHYgjFVearvb2dPHAgKbLcGNJmS77cTEerhUZx3/mVhVjM2bSLq18eXMGABLhmWZLyUxJoqSvnhXOnOLx/O7OcHX5wfNn5VNu9StsC1rAxJsEqJ9qHrtwp7GpNY011LLFT3W47nhQfQVdbLW0NQkZ5LuWFBurn5mERC1cnz6JIyr7VXa18/cV7/Pn/3uPmtcs8f+ow777zKt72WnfWj9Kyx60fpTVs/GQJ9nWlzRjI+tp4DvfkUprof1vu7Ow0kTop6lf0tFImQM0FRiqLTGTHRVA+xZ+q9AS293ZI6Xsd+BK++IADG9ewY1M/Nw4O/ahnkD1utbGvtC1gDRs/SZwcHahPC2R5aSiPdGaxrjYRt4m3u6DvJC8Gly1S1rWutXWleRjCp5MR6Ms8SS13STTfOtjHbqmUbl19nsP7dtPfu4CrUh4aXPWKBf2gJbs4+kFLw8Ydi7JicZwP3fnBnOgr5Pn1lQRPuv2ejhMmUFWaT5fM26qSLJqqS6iYGki2hytz0pOoMKWxqKyAtQ/OpygrlXJleUMy61cv5tG+Rdjb2f1oC8vWs76FNWzcsYQFONORHcTFwbns6zBRnjLVpgwsyE4jIV4tS1mYCzMxStyImxZEuYCryk5lcYuF6xefBODJPVvoqS6lVdx/aUYcJRP1at/Ro7S8PtKP0ho27kh8XB1oMShXDuPSxirW1yUy2ev2+jpGsqua8gJSkuIpjAijwMOFBJm39eZssaSReouZi0/sZ+Tnm68+Zvf2QczhQfpjj148qJd1+sWDho07KP3uoSrJh46cKexbZOR4XwGpIbffa8a0KSxqqlRJgyQfeTSKK88rlxJQwJoMCZQV50tWdRjeuwoAIHP2GZ57+ji8cBrfCT/sytotHtmY1N/i0bDxT0thtJfVlQdrrFGZskQ/JtiP2IGMDmdZ13xJLszMToihICGaegHbJJVQeb5RrJvL5WeH+fL6Ba48uovzxw7IJsBGDu5ezwf7t9Ic4q8Z88cFLXktqx+0NGz8UxLq70ybKZDFEqjOD5TTlDkdpxHWmDl9Kj0L6yjNN1jnbHVuBtFBgVgkZ04V8HmSfJwe3sX/XjvLExuWc+mpg6xc0kavZFpX1iyhYPKd7cJImqyMqF88aNj40eLm5EBdii/tWVM4vaKIVVXxeI7IgoJkXR9Y3k61OZc5WWmY8zOw5BrJlCQnPMifzJgIultrOXtoK48MLOXQllUs65gngauabZZS8v3uPNUVLGrl0bewho0fLZlRfizMDGR1RbRYt4y4YPfvyHB1YZksLbNioynOMZAhmVParEjMmalkJcaRY0pl/fIOTu5ey6PbBzh+YBNd8yvYY85il7h7vqzJXu53/iZEXsCrrgOlbQFr2PjBlC06bAZRM6fRlj2NB/OnSXIxi6rkAAle/6iSHCWLamZWXDQFkgvHx4QTI4lFUmgIMXJdWZ6JobU9vH7hCEPrV9Aj7ttWW0qT5NPbSky0BQX85ARIug5Uq4XStoA1bIwqKiUsNCRTmJtOdkwo6aFeEqD8KYqfZN16tTIr5yzvFrCx4YQLwLjoMJJk6fHx9iAqdBqV4tqb2hs5vmcd7fUVdDdX0yr1cZ1YdZWszWHuY1OTS6uF6i9R2hawho1RC4I5xhTaZc797csPOffkMMa4cIJlnoWFzlAZkNWyVeY8LBJ9I8KmC+gIARtO7IxgwqdPoU4i8mXZPB8eWktXY6UAnivRupDFUkSsTIhW9fOYVWvSX6KaapS2BaxhQ58xHy9KstI5eewg8Dc+//gd1q3qpVDmYrokEn6TvImJDCEtKpTQyQEkinWTQqYROSWAsMl+FBiSeOHso5w9NkSLpYymyhI6m2rZb0glSvMWfyxEmmpUJ5HStoA1bOhK9Mzpso6Ws2lgGX+4+QqbB5bT0VJPXbWZXKlXvb08iImaQYwkGFHTJjNFCIiQ5afcmET3/GqxdCTnTjws87RE5msFFnMhve7Oqj6+O283XB1V+5R+F4+GDV2Z6OCAJT+XVUu76F5QT01ZoczlDHJmx5MoZETI/IyQgBTs70uAlyfZ4qLzck3skP6MocHlhMkx9UK7c34t1eEz8J+gLeLHHrDUB/qANWzoVkHTfHyY5u1NRWoSG1b30dFYS5kxjfy4GPKzU5glQczTw43wqZOx5Bi5deUsjw1tZnD5Ytat6CZJzjMG+tHl6YaH1qpjL6pnTDXKKW0LWMOGjcz0c6I6JZqYoCBmBvhiDAsjNSyE5OBgirPTMaXEMTPQnxCx7taV3bz/2nkO7VrPmaN7JbFooLPBwkaZ0wFOmrl6ly0s3YH6FtawcZu4TrTHkuRHWWwo25syKJw9nZBJTtZjfp4eVGWkYgoLJU/m8a3zR/nLW8+zd8NKuprrWdHZwsaVXTwpc9bewR47jWXvpkh3oPJapW0Ba9j4VuxlKSqK9eLBvKlsnZfI4905LMwOVu1E1uOJ8RKIju3hyI4BKXLO8q4UAKva59PT1iigV3B611oK/r7bIYD/dWBFVEuk6gNV2hawho1vJTLQmVaDP90F0xmWKmhTfYLM5e/cckVPJ3z1R14+c5jn1vWzuXM+2xa3cf3FE6yV6sg4wf5fClKbaYnX6mdaGjas4u5sb3212Z4ZxIbaeA51ZZIX7X3b643Nq5fC52+wu7+HLd0LefXhDXzyyjk2GpKZ7PDzgRVRza9qqVXaFrCGDatkR3jQnO5HmzHAWtR3z5mJq+PtIHZLk9hXt15kW2877597nKNLWhmSwsBhHLRNSMevanNW2hawhg0muTkosCL+LCsJZUtDEsE+thH2pszbt47t5Z3j+zizbikzNanhzynS5qx6u5W2BTySjchAFxZkBlsBt2cHsbNZonCEfsp5/coZfn9+mJcHeoj5e3AaNyJpsmpo1y8eRrIRF+RCY6qvuLI//WURLMoPk7JP/83gxiITa9Kixg1IrUtLF7++S49kwxTmQUuGZER5wSwpDsPje97jeDvYqd6NcQlYuvjVNFXaFvAINsbtjzVE1N9K1HN9+1MFeS714xT1zN9+JyDV/2qp1U8tR7AxLn+sIVq1N6uebjWG6t5XP1lQY6pnGPUceW71uw1bwJofa/xbWFjwfLeD89/Pf8jn/wEkLDAzA7CwbwAAAABJRU5ErkJggg==); height: 60px; width: 60px; background-repeat: no-repeat; display: inline-block;"
  />
</div>
`;
